### PR TITLE
Implement leader forwarding in the Cluster Agent (remove the redirect in reply)

### PR DESCRIPTION
**Pull request by vboulineau**
### What does this PR do?

This PR adds a reverse proxy to Cluster Agent followers, transparently forwarding queries to Cluster Agent leader if necessary.
It also changes the way the Agent connects to the Cluster Agent:
- Only 1 connection instead of 2 currently with endpoint checks)
- Re-connect every 20 minutes (by default) to ensure connections are properly spread in case of Cluster Agent scale-up or scale-down.

It allows for better scalability of the Cluster Agent when adding more Cluster Agent replicas.

### Motivation

Performance.

### Additional Notes

Change is backward and forward compatible:
- Agent >= 7.37 and Cluster Agent >= 1.21: Follower forwarding is used
- Agent < 7.37 and Cluster Agent >= 1.21: Follower forwarding is used
- Agent >= 7.37 and Cluster Agent < 1.21: Follower forwarding is not used, a warning is logged.

The Cluster Agent will answer with `502 Bad Gateway` when leader does not answer.
Connections between each follower and leader are bounded to `500` by default.
Connections will be re-used as much as possible.

### Possible Drawbacks / Trade-offs

Leader state handling in the Cluster Agent HTTP Server may come time after `is_leader: true` has been set on metrics. Meaning that you may see `datadog.cluster_agent.api_requests` metric tagged with both `is_leader:true` and `forwarded:true`.
This is expected, the query will fail (answered with 502).

### Describe how to test/QA your changes

Mostly non-regression testing: none of the feature should be impacted, the change should not be visible.
The `datadog.cluster_agent.api_requests` has been modified to include a `forwarded tag`.
A notebook is available to check the Cluster Agent queries.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by vickenty — APPROVED** [review]

File: releasenotes-dca/notes/cluster-agent-forwarding-23ca730171446dc1.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+enhancements:
+  - |
+    The Cluster Agent followers now forward quries to Cluster Agent leaders themselves, allowing to reduce overall number of connection to the Cluster Agent and to better spread the load between leader and forwarders.
```
Comment (cswatt) [inline releasenotes-dca/notes/cluster-agent-forwarding-23ca730171446dc1.yaml:4]:
```suggestion
    The Cluster Agent followers now forward queries to the Cluster Agent leaders themselves. This allows a reduction in the overall number of connections to the Cluster Agent and better spreads the load between leader and forwarders.
```

File: releasenotes/notes/cluster-agent-forwarding-97f012dbec094401.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+enhancements:
+  - |
+    The Cluster Agent followers now forward quries to Cluster Agent leaders themselves, allowing to reduce overall number of connection to the Cluster Agent and to better spread the load between leader and forwarders.
```
Comment (cswatt) [inline releasenotes/notes/cluster-agent-forwarding-97f012dbec094401.yaml:4]:
```suggestion
    The Cluster Agent followers now forward queries to the Cluster Agent leaders themselves. This allows a reduction in the overall number of connections to the Cluster Agent and better spreads the load between leader and forwarders.
```

**Review by cswatt — APPROVED** [review]
left grammar check on release notes. approving PR so as not to block.

**Review by ahmed-mez — APPROVED** [review]
Great improvement!
Let's make sure @DataDog/integrations-tools-and-libraries do the QA on cloud foundry as well.

**Pull request merged**